iml-agent/src/action_plugins/stratagem/action_purge.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2019 DDN. All rights reserved.
```

Comment:
I've updated this file a little bit to use futures all the way through the action (which is a requirement for a responsive runtime). I did this mostly by using the blocking api (https://docs.rs/tokio-threadpool/0.1.14/tokio_threadpool/fn.blocking.html). In addition, I used a spawn call for each rmfid. Let me know if this looks ok to you:

```rust
// Copyright (c) 2019 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

use crate::{agent_error::ImlAgentError, fidlist, http_comms::mailbox_client};
use futures::{future::poll_fn, Future, Stream};
use tokio_threadpool::blocking;

pub fn purge_files(
    device: &str,
    args: impl IntoIterator<Item = String>,
) -> Result<(), ImlAgentError> {
    liblustreapi::rmfids(&device, args).map_err(ImlAgentError::LiblustreError)
}

fn search_rootpath(device: String) -> impl Future<Item = String, Error = ImlAgentError> {
    poll_fn(move || {
        blocking(|| liblustreapi::search_rootpath(&device))
            .map_err(|_| panic!("the threadpool shut down"))
    })
    .and_then(std::convert::identity)
    .from_err()
}

fn rm_fid(mntpt: String, fid: String) -> impl Future<Item = (), Error = ImlAgentError> {
    poll_fn(move || {
        blocking(|| liblustreapi::rmfid(&mntpt, &fid))
            .map_err(|_| panic!("the threadpool shut down"))
    })
    .and_then(std::convert::identity)
    .from_err()
}

pub fn read_mailbox(
    (device, mailbox): (String, String),
) -> impl Future<Item = (), Error = ImlAgentError> {
    search_rootpath(device).and_then(move |mntpt| {
        mailbox_client::get(mailbox)
            .and_then(|s| serde_json::from_str(&s).map_err(ImlAgentError::Serde))
            .for_each(move |fli: fidlist::FidListItem| {
                tokio::spawn(
                    rm_fid(mntpt.clone(), fli.fid)
                        .map_err(|e| log::warn!("Error removing fid {:?}", e)),
                );
                Ok(())
            })
    })
}
```